components/usage/pkg/apiv1/usage.go

```diff
@@ -136,6 +136,36 @@ func (s *UsageService) ReconcileUsage(ctx context.Context, req *v1.ReconcileUsag
 
 }
 
+func (s *UsageService) GetCostCenter(ctx context.Context, in *v1.GetCostCenterRequest) (*v1.GetCostCenterResponse, error) {
+	var attributionIdReq string
+
+	if in.AttributionId != "" {
+		attributionIdReq = in.AttributionId
+	}
+
+	if in.AttributionId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Empty attributionId")
+	}
+
+	attributionId, err := db.ParseAttributionID(attributionIdReq)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to parse attribution ID: %w", err)
+	}
+
+	result, err := db.GetCostCenter(ctx, s.conn, db.AttributionID(attributionIdReq))
+
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Cost center not found: %s", err)
```

Comment:
Here, you'll need to do the `errors.Is(db.CostCenterNotFoundError, err)` check and only return NotFound if the check evalutes to true, otherwise return `codes.Internal` for the status code